### PR TITLE
Add documentation for feature support, usage, and caveats

### PR DESCRIPTION
**Pull request by achow101**
Adds documentation for devices supported and features supported. Add some usage instructions and some special notes about the commands for each device.

### CODE REVIEW

**Comment by achow101** [comment]
@instagibbs Please review the ledger docs and feature support in the table. Some things are probably wrong as I only added things that I knew of.

**Comment by instagibbs** [comment]
Will do.

File: docs/ledger.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+* `getmasterxpub`
+* `signtx` (with some caveats)
+* `getxpub`
+* `signmessage`
+
+## `signtx` Caveats
+
+Due to device limitiations, not all kinds of transactions can be signed by a Ledger. 
+
+* Transactions containing both segwit and non-segwit inputs are not supported
```
Comment (instagibbs) [inline docs/ledger.md]: "Transactions containing both segwit and non-segwit inputs are not entirely supported; only the segwit inputs will be signed in this case."

File: README.md [hunk 2]
```diff
@@ -2,9 +2,60 @@
...
-## Using
+## Prerequisites
 
-You will need Python 3. The main script is `hwi.py`.
+Python 3 is required. The libraries and udev rules for each device must also be installed.
+
+Install all of the libraries using `pip`:
+
+```
+sudo pip3 install hidapi # HID API needed in general
```
Comment (instagibbs) [inline README.md]: or in a virtualenv

**Comment by instagibbs** [comment]
ACK ledger parts and general description

**Pull request merged**